Cython/Utility/CppConvert.pyx

```diff
@@ -40,12 +40,22 @@ cdef inline object {{cname.replace("PyObject", py_type, 1)}}(const string& s):
 cdef extern from *:
     cdef cppclass vector "std::vector" [T]:
         void push_back(T&) except +
+        void reserve(size_t) except +
+
+    cdef Py_ssize_t PyObject_LengthHint(object o, Py_ssize_t defaultval)
```

Comment:
```suggestion
    cdef Py_ssize_t PyObject_LengthHint(object o, Py_ssize_t defaultval) except -1
```

Reply:
I think we do want to keep the `except -1` here.

It's not that it should succeed to `PyObject_LengthHint` throws an exception - it's just that the failure should be tested.

Reply:
Why shouldn't it succeed? We're just silently getting a hint from an object to improve performance. If getting the hint raises, we fall back and nothing prevents the conversion code from working anyway.

Reply:
That could be a reasonable choice. In which case the solution is to clear the exception rather than to ignore it.